File: TandemBike/tandem.py
def tandemBicycle(redShirtSpeeds, blueShirtSpeeds, fastest):
    
    maxSpeed = 0
    redShirtSpeeds.sort()
    blueShirtSpeeds.sort()
    
    if not fastest:
        reverseArrayInPlace(redShirtSpeeds)
        
    for i in range(len(redShirtSpeeds)):
        rider1 = redShirtSpeeds[i]
        rider2 = blueShirtSpeeds[len(blueShirtSpeeds) - i - 1]
        maxSpeed += max(rider1, rider2)
        
    return maxSpeed


def reverseArrayInPlace(arr):
    start = 0
    end = len(arr) - 1
    while start < end:
        arr[start], arr[end] = arr[end], arr[start]
        start += 1
        end -= 1

red = [1,3,5,7,8]
blue = [4,1,6,8,1]

print(tandemBicycle(red, blue, False))

print(tandemBicycle(red, blue, True))

File: TandemBike/tandem.js
function tandemBicycle(redShirtSpeeds, blueShirtSpeeds, fastest) {
    // Write your code here.
    let maxSpeed = 0;
    redShirtSpeeds.sort((a, b) => a - b);
    blueShirtSpeeds.sort((a, b) => a - b);

    if (!fastest) {
        reverseArray(redShirtSpeeds);
    }

    for (let i = 0; i < redShirtSpeeds.length; i++) {
        const rider1 = redShirtSpeeds[i];
        const rider2 = blueShirtSpeeds[blueShirtSpeeds.length - i - 1];
        maxSpeed += Math.max(rider1, rider2);
    }

    return maxSpeed;
}

function reverseArray(arr) {
    let start = 0;
    let end = arr.length - 1;
    while (start < end) {
        const temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp;
        start++;
        end--;
    }
}

let red = [1,3,5,7,8];
let blue = [4,1,6,8,1];

console.log(tandemBicycle(red, blue, true));

console.log(tandemBicycle(red, blue, false));